### PR TITLE
Check if the runpath exists before starting simulations

### LINKED ISSUE #4182: Warn user if running in existing run path
Sometimes this is a wanted feature, but it can also lead to strange side effects, we should warn users when they are trying to run in an existing run path, so they are aware of the potential problems.

### PR DESCRIPTION
**Pull request by frode-aarstad**
**Issue**
Resolves #4182

**Approach**
_Short description of the approach_

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/shared/models/base_run_model.py [hunk 1]
```diff
@@ -525,3 +525,21 @@ def _post_update_data(self, parent_ensemble_id: str, algorithm: str) -> str:
         )
         self.setPhaseName("Uploading done")
         return update_id
+
+    def check_if_runpath_exists(self) -> bool:
+        run_path = self.facade.run_path
+        start_iteration = 0
+        if "start_iteration" in self._simulation_arguments:
```
Comment (ManInFez) [inline src/ert/shared/models/base_run_model.py:532]:
```suggestion
        start_iteration = self._simulation_arguments.get("start_iteration", 0)
```
```

**Comment by ManInFez** [comment]
This needs a rebase and tests to go green, otherwise just a small comment

File: src/ert/gui/simulation/simulation_panel.py [hunk 2]
```diff
@@ -123,31 +123,53 @@ def runSimulation(self):
...
+            )
+
+            abort = False
+            if model.check_if_runpath_exists():
+                if (
+                    QMessageBox.warning(
+                        self,
+                        "Start simulations?",
+                        (
+                            "ERT is running in an existing runpath.\n\n"
```
Comment (oyvindeide) [inline src/ert/gui/simulation/simulation_panel.py:147]: Could we also mention how many of the realizations are running in existing run paths? I think a common way to run us to use `Single realization test-run` followed by a full ensemble. So in that case say: `1 out of N realizations are running in existing run paths`

**Review by oyvindeide — APPROVED** [review]
Looks very good, just a small comment

File: src/ert/gui/simulation/simulation_panel.py (hunk 2 above)
Comment (frode-aarstad) [reply src/ert/gui/simulation/simulation_panel.py:147]: So in this case N is ensemble size not active mask size?

Comment (frode-aarstad) [reply src/ert/gui/simulation/simulation_panel.py:147]: Will not implement this change in this PR as it requires quite some work to get the correct number of realizations. Created a new issue just for this. #4410

**Pull request merged**